### PR TITLE
[fix]Encoding with saving report to files in Windows

### PR DESCRIPTION
**Pull request by simson2010**
## Fix 
 - Fix incorrect encoding in Windows when saving report to disk.
 
## Add
 - add new ignore folder "flagged" in gitignore file.

### CODE REVIEW

File: src/report_generator.py [hunk 1]
```diff
@@ -50,9 +50,9 @@ def generate_daily_report(self, markdown_file_path):
         report = self.llm.generate_daily_report(markdown_content)  # 调用LLM生成报告
 
         report_file_path = os.path.splitext(markdown_file_path)[0] + "_report.md"
-        with open(report_file_path, 'w+') as report_file:
+        with open(report_file_path, 'w+', encoding='utf-8') as report_file:
```
Comment (nanshaqundao) [inline src/report_generator.py:53]: This is proved to be working, without the fix when running in windows we get
`Lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeEncodeError: 'charmap' codec can't encode characters in position 7-10: character maps to <undefined>`

**Comment by nanshaqundao** [comment]
image
I kinda get why DjangoPeng not having this issue, as he is using macOS which by default use utf-8 for file operation encoding.
For windows, if update the locale as above attached, then problem is solved - no need to specify the encoding during `open` operation.

**Comment by DjangoPeng** [comment]
Please fix conflicts then we can move on @simson2010 @nanshaqundao

**Comment by nanshaqundao** [comment]
I think the issue can be closed by just setting at OS level the locale config ( enabling utf-8)

Probably worth update the readme file mentioning it.

Regards,

Nan Sha

________________________________
发件人: Jingtian Peng ***@***.***>
发送时间: Wednesday, September 18, 2024 7:23:19 AM
收件人: DjangoPeng/GitHubSentinel ***@***.***>
抄送: nanshaqundao ***@***.***>; Mention ***@***.***>
主题: Re: [DjangoPeng/GitHubSentinel] [fix]Encoding with saving report to files in Windows (PR #12)


Please fix conflicts then we can move on @simson2010<https://github.com/simson2010> @nanshaqundao<https://github.com/nanshaqundao>

―

**Comment by DjangoPeng** [comment]
Closed

**Pull request closed without merging**